metadata/src/main/java/org/apache/kafka/metadata/authorizer/StandardAuthorizerData.java

```diff
@@ -437,14 +436,14 @@ private void checkSection(
     /**
      * The set of operations which imply DESCRIBE permission, when used in an ALLOW acl.
      */
-    private static final Set<AclOperation> IMPLIES_DESCRIBE = Collections.unmodifiableSet(
-        EnumSet.of(DESCRIBE, READ, WRITE, DELETE, ALTER));
+    private static final Set<AclOperation> IMPLIES_DESCRIBE =
+        EnumSet.of(DESCRIBE, READ, WRITE, DELETE, ALTER);
 
     /**
      * The set of operations which imply DESCRIBE_CONFIGS permission, when used in an ALLOW acl.
      */
-    private static final Set<AclOperation> IMPLIES_DESCRIBE_CONFIGS = Collections.unmodifiableSet(
-        EnumSet.of(DESCRIBE_CONFIGS, ALTER_CONFIGS));
+    private static final Set<AclOperation> IMPLIES_DESCRIBE_CONFIGS =
+        EnumSet.of(DESCRIBE_CONFIGS, ALTER_CONFIGS);
```

Comment:
the return set of `EnumSet.of` is not immutable. Maybe we can use `Set.of` instead?